shuffle/cli/context.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import * as path from "https://deno.land/std@0.110.0/path/mod.ts";
+import urlcat from "https://deno.land/x/urlcat@v2.0.4/src/index.ts";
+import { green } from "https://deno.land/x/nanocolors@0.1.12/mod.ts";
+import { isURL } from "https://deno.land/x/is_url@v1.0.1/mod.ts";
+
+export const shuffleDir = String(Deno.env.get("SHUFFLE_HOME"));
+export const projectPath = String(Deno.env.get("PROJECT_PATH"));
+export const nodeUrl = getNetworkEndpoint(
+  String(Deno.env.get("SHUFFLE_NETWORK")),
+);
+export const senderAddress = String(Deno.env.get("SENDER_ADDRESS"));
+export const privateKeyPath = String(Deno.env.get("PRIVATE_KEY_PATH"));
+export const privateKeyBytes = normalizePrivateKey(
+  await Deno.readFile(privateKeyPath),
+);
+
+export const receiverPrivateKeyPath = path.join(
+  shuffleDir,
+  "accounts/test/dev.key",
+);
+export const receiverAddressPath = path.join(
+  shuffleDir,
+  "accounts/test/address",
+);
+export const receiverAddress = await Deno.readTextFile(receiverAddressPath);
+
+function highlight(content: string) {
+  return green(content);
+}
+
+console.log(`Loading Project ${highlight(projectPath)}`);
+console.log(`Connected to Node ${highlight(nodeUrl)}`);
+console.log(`Sender Account Address ${highlight(senderAddress)}`);
+console.log(
+  `"shuffle", "main", "codegen", "helpers" top level objects available`,
+);
+console.log(`Run "help" for more information on top level objects`);
+console.log();
+
+export function addressOrDefault(addr: string | undefined): string {
+  if (addr) {
+    return addr;
+  }
+  return senderAddress;
+}
+
+function getNetworkEndpoint(inputNetwork: string) {
+  if (inputNetwork == "unknown") {
+    throw new Error("Invalid network.");
+  }
+  let network = "";
+  if (isURL(inputNetwork)) {
+    network = inputNetwork;
+  } else {
+    network = urlcat("http://", inputNetwork);
+  }
+  return network;
+}
+
+function normalizePrivateKey(privateKeyBytes: Uint8Array): Uint8Array {
```

Comment:
TL;DR: I believe it's an artifact of BCS encoding the raw private key bytes leading to a leading " " white space padding the private key from 32 bytes to 33.

Details:
I did some more digging into this to give you a better answer. For some reason, all of the keys we are consuming, including `nodeconfig/mint.key` and `latest/dev.key` have a space as a prefix before the actual private key. This is as a result of writing and read keys in raw bcs bytes, rather than hex encoded bytes:

https://github.com/dimroc/diem/blob/shuffle-ts-refactor/config/generate-key/src/lib.rs#L39-L43

I'm not sure why it is written out that way, but when reading, this accommodate for that peculiarity to get the dev up and running.

![CleanShot 2021-11-05 at 12 33 51@2x](https://user-images.githubusercontent.com/635121/140568382-339c32c2-6043-4df6-9920-0bc42b8c4c49.png)
(end obfuscated)

![CleanShot 2021-11-05 at 12 44 15@2x](https://user-images.githubusercontent.com/635121/140569597-981cad45-1cd2-4aaa-8e43-50d6c85b3b6f.png)


If anything, this seems like a bug on the write.